polaris-react/src/components/Frame/components/Toast/Toast.tsx

```diff
@@ -67,20 +70,59 @@ export function Toast({
     </div>
   ) : null;
 
-  const leadingIconMarkup = error ? (
-    <div className={styles.LeadingIcon}>
-      <Icon source={AlertCircleIcon} tone="inherit" />
-    </div>
-  ) : null;
+  let leadingIconMarkup;
```

Comment:
It actually makes readability more messy due to typescript:

With the following:
```
const leadingIconMarkup =
  error || icon ? (
    <div className={styles.LeadingIcon}>
      <Icon source={error ? AlertCircleIcon : icon} tone="inherit" />
    </div>
  ) : null;
```

Typescript will complain that the `source` prop cannot be undefined. That's because it is unable to pickup the logic where `source=icon` only if `error` is nullish and that means that `icon` is defined. The solution for this involves adding some additional conditionals, and that makes readability really bad because we get multiple levels of nested ternary. I'd rather have a `let` with 2 conditional, a single level of nesting and a bit of repetition.